### DISCUSSION TITLE
Simplifying the parsing

### DISCUSSION THREAD

**Discussion by pintariching** [pintariching/rustle#52] [Q&A]
I think the original compiler goes in a big roundabout way when parsing. Couldn't we simplify it?
I think we can start by splitting the `.svelte` file into three parts. Extract the script and style and the remainder is svelte flavored HTML, where all the parts between curly braces get thrown through the swc parser.

**Comment by qwertydelle** [comment]
I think this what the svelte compiler already does but instead of `swc` it uses `acorn`. The tag.ts file basically has special cases for `style` tags and `script` and every other tag us treated as an html tag. We could use nom to simplifying parsing but at this point we are too far ahead.

**Reply by pintariching** [reply]
I don't think we're too far ahead, I plan to rewrite the parsing anyway when I figure out how it works. How could we use `nom`?

**Reply by qwertydelle** [reply]
Nom is a parser combinator framework basically it breaks your parser components into small chunks. It is also declarative so much easier to work on in the future. But what I think once we finish the current parser it would much easier to implement it in nom in the future.

**Reply by qwertydelle** [reply]
Also for the current parser all we really need to implement so far is the mustache and tag functions.

**Reply by mystique09** [reply]
What about pest.rs for parsing?

**Comment by pintariching** [comment]
I made a branch [rustify-parsing](https://github.com/pintariching/rustle/tree/rustify-parsing) where I followed the awesome video from [Tan Li Hau](https://github.com/tanhauhau) called [Build your own Svelte](https://www.youtube.com/watch?v=mwvyKGw2CzU). He goes into detail on how it works and I think I now get it how it works. He also released a [followup](https://www.youtube.com/watch?v=5zR1F7NiLUU) on implementing more features.

**Reply by qwertydelle** [reply]
Oh wait so is parsing complete?

**Reply by pintariching** [reply]
Not really complete, just some of it works. Check the integration tests, I have an example. A lot of functionality is still missing but it can parse a simple file now!

**Reply by qwertydelle** [reply]
Nice project going good

**Reply by pintariching** [reply]
I commented out a lot of the code we've written so far for now. I'm going to try and follow the video to get a simple compiler going. Currently I'm stuck at the analyze stage where in the video he uses a library 'periscopic' to traverse the AST and split the code into scopes. I think SWC has this functionality but the documentation is a bit vague.
